src/core/gltf2importer/gltf2exporter.cpp

```diff
@@ -0,0 +1,372 @@
+
+            auto compressed_mesh = compressPrimitive(
+                    primitive_json,
+                    primitives[p]);
+
+            if (!compressed_mesh.primitive_json.empty()) {
+                //* all the side-effects of this method are in this block: */
+                primitives_json[p] = compressed_mesh.primitive_json;
+                compressedBufferData.push_back(compressed_mesh.compressed_data);
+                bufferViews.push_back(compressed_mesh.new_buffer_view);
+                accessorsToClean.insert(compressed_mesh.accessors_to_clean.begin(), compressed_mesh.accessors_to_clean.end());
+
+                newBufferViewIndex++;
+            } else {
+                const auto meshName = mesh_json[GLTF2Import::KEY_NAME].toString().toUtf8();
+                if (!meshName.isEmpty())
+                    qCWarning(kuesa, "A mesh could not be compressed: %s -> %d", meshName.constData(), p);
+                else
+                    qCWarning(kuesa, "A mesh could not be compressed: %d -> %d", mesh_idx, p);
+            }
+        }
+
+        mesh_json[GLTF2Import::KEY_PRIMITIVES] = std::move(primitives_json);
+        return mesh_json;
+    }
+
+    // Compress a single primitive of a mesh
+    CompressedGLTFPrimitive compressPrimitive(
+            QJsonObject primitive_json, const Kuesa::GLTF2Import::Primitive &primitive) const
+    {
+        CompressedGLTFPrimitive compressed_primitive;
+
+        // Do the compression
+        const auto compressed = Kuesa::Compression::compressMesh(*primitive.primitiveRenderer->geometry());
+        if (!compressed.buffer)
+            return {};
+
+        int offset = compressedBufferData.size();
+        auto &eb = *compressed.buffer.get();
+        int eb_size = static_cast<int>(eb.size());
+
+        compressed_primitive.compressed_data = QByteArray { eb.data(), eb_size };
+
+        // For now we allocate new bufferViews per compressed chunk; then we should do a pass to remove / replace unused bv ?
+
+        // Allocate a new buffer view
+        {
+            compressed_primitive.new_buffer_view[GLTF2Import::KEY_BUFFER] = compressedBufferIndex;
+            compressed_primitive.new_buffer_view[GLTF2Import::KEY_BYTEOFFSET] = offset;
+            compressed_primitive.new_buffer_view[GLTF2Import::KEY_BYTELENGTH] = eb_size;
+        }
+
+        // Create or modify the extension object
+        {
+            auto ext_obj = primitive_json[GLTF2Import::KEY_EXTENSIONS].toObject();
+
+            {
+                QJsonObject draco_ext;
+                draco_ext[GLTF2Import::KEY_BUFFERVIEW] = newBufferViewIndex;
+                {
+                    QJsonObject draco_ext_attr;
+                    for (const auto &[name, idx] : compressed.attributes) {
+
+                        // TODO the names aren't the correct ones, how to get them ?
+                        // e.g. vertexNormal instead of NORMAL
+                        // for the sake of testing, this dirty hack :
```

Comment:
QAttribute::defaultPositionAttributeName, QAttribute::defaultNormalAttributeName, QAttribute::defaultColorAttributeName, QAttribute::defaultTextureCoordinateAttributeName, QAttribute::defaultTextureCoordinate1AttributeName, QAttribute::defaultTextureCoordinate2AttributeName,
QAttribute::defaultTangentAttributeName, QAttribute::defaultJointIndicesAttributeName, QAttribute::defaultJointWeightsAttributeName

Reply:
I should maybe squash my commits so that there is only the last version since there has been a bunch of back-and-forth,  what do you think ?

Reply:
Actually that's fine. I found I could use the Files Changed tab to do the review which gives me the same as if all commits had been squashed. Please ignore comments which aren't relevant any more.